app/workers/post_process_media_worker.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class PostProcessMediaWorker
+  include Sidekiq::Worker
+
+  def perform(media_attachment_id)
+    media_attachment = MediaAttachment.find(media_attachment_id)
+    media_attachment.processing = :in_progress
+    media_attachment.save
+    media_attachment.file.reprocess!
+    media_attachment.processing = :complete
+    media_attachment.save
+  rescue Paperclip::Error
```

Comment:
I wonder if there are possibly other exceptions we might run into. Also it may make sense to log *why* processing failed somewhere.

Reply:
Validation happens in the request/response cycle, so there should be very little unexpected here. Errors that wouldn't fit under Paperclip::Error that I can think of would be disk-related (not writeable, directory doesn't exist) or network related (S3 timeout)

If you mean logging to the Rails log, I'm for it, but I wouldn't go to saving the exception in a special column, I think that would be overkill and not really useful.

The question is do we bother to retry these on failures? I think most such failures would likely not recover within seconds, and that means it's already keeping the end-user waiting too long, so maybe best to communicate failure to the UI somehow.

Reply:
Actually one thing to watch out for is that case when a converted file is bigger than the uploaded one. I don't think it happens too much with media attachments though, it's mostly when a GIF is converted to GIF, on avatars and headers. A video should never be bigger than a GIF to the point of exceeding the limit, I think...